sycl/source/detail/device_impl.cpp

```diff
@@ -74,7 +74,7 @@ device_impl::device_impl(pi_native_handle InteropDeviceHandle,
 }
 
 device_impl::~device_impl() {
-  if (!MIsRootDevice && !MIsHostDevice) {
```

Comment:
Looks like the failure was caused by your change:
/localdisk2/sycl_ci/buildbot/worker/sycl-ubu-x64-pr/llvm.obj/tools/sycl/unittests/scheduler/./SchedulerTests(+0xb6d5a)[0x556a1871fd5a]
/localdisk2/sycl_ci/buildbot/worker/sycl-ubu-x64-pr/llvm.obj/tools/sycl/unittests/scheduler/./SchedulerTests(+0xb4f64)[0x556a1871df64]
/localdisk2/sycl_ci/buildbot/worker/sycl-ubu-x64-pr/llvm.obj/tools/sycl/unittests/scheduler/./SchedulerTests(+0xb50b3)[0x556a1871e0b3]
/lib/x86_64-linux-gnu/libpthread.so.0(+0x12890)[0x7f8d30ed7890]
/rdrive/ref/opencl/runtime/linux/oclgpu/20.21.16886/libze_intel_gpu.so.0.8(_ZN2L014CommandQueueHwIL14GFXCORE_FAMILY12EE7destroyEv+0x25)[0x7f8d2ef43375]
/rdrive/ref/opencl/runtime/linux/oclgpu/20.21.16886/libze_intel_gpu.so.0.8(_ZN2L011CommandListD1Ev+0x30)[0x7f8d2ef49890]
/rdrive/ref/opencl/runtime/linux/oclgpu/20.21.16886/libze_intel_gpu.so.0.8(_ZN2L033CommandListImmediateProductFamilyIL14PRODUCT_FAMILY20EED0Ev+0x17)[0x7f8d2ef3ee77]
/rdrive/ref/opencl/runtime/linux/oclgpu/20.21.16886/libze_intel_gpu.so.0.8(_ZN2L014CommandListImp7destroyEv+0x13)[0x7f8d2ef49c13]
**/localdisk2/sycl_ci/buildbot/worker/sycl-ubu-x64-pr/llvm.obj/./lib/libpi_level0.so(piDeviceRelease+0x39)[0x7f8d2f741cc9]**
/localdisk2/sycl_ci/buildbot/worker/sycl-ubu-x64-pr/llvm.obj/tools/sycl/unittests/scheduler/./SchedulerTests(+0x20f75)[0x556a18689f75]
/localdisk2/sycl_ci/buildbot/worker/sycl-ubu-x64-pr/llvm.obj/tools/sycl/unittests/scheduler/./SchedulerTests(+0x1ff6a)[0x556a18688f6a]
/localdisk2/sycl_ci/buildbot/worker/sycl-ubu-x64-pr/llvm.obj/tools/sycl/unittests/scheduler/./SchedulerTests(+0x9baea)[0x556a18704aea]
/lib/x86_64-linux-gnu/libc.so.6(+0x43041)[0x7f8d2fba8041]
/lib/x86_64-linux-gnu/libc.so.6(+0x4313a)[0x7f8d2fba813a]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xee)[0x7f8d2fb86b9e]
/localdisk2/sycl_ci/buildbot/worker/sycl-ubu-x64-pr/llvm.obj/tools/sycl/unittests/scheduler/./SchedulerTests(+0x15c3a)[0x556a1867ec3a]

Reply:
This is expected failure due to the issue in L0 RT.
The L0 RT issue is now fixed, and we are still waiting for their release (soon).
I will rerun this failure once the new L0 RT is released.